### PR TITLE
DR-2610 Fix Oauth Timeout with refresh token

### PR DESCRIPTION
**Pull request by nmalfroy**
This is an example of how refreshing access tokens via a refresh token could work.

I've deployed this on:
https://jade-nm.datarepo-dev.broadinstitute.org

### CODE REVIEW

**Comment by dinvlad** [comment]
I think we need to add removal of the token to the existing logic dealing with explicit/15-minute logout also.

**Comment by nmalfroy** [comment]
Ah interesting.  I think it does an explicit log out so it clears out the refresh token but I can confirm

**Comment by dinvlad** [comment]
Oh I see - the entire "user data" config (including refresh token) is removed upon logout right?

File: src/index.jsx [hunk 1]
```diff
@@ -103,27 +103,38 @@ function bootstrap() {
...
     prompt: 'login',
     scope: scopes.join(' '),
+    loadUserInfo: isGoogleAuthority,
     userStore: new WebStorageStateStore({ store: window.localStorage }),
+    extraQueryParams: {'access_type': 'offline'},
   };
 
+  // Log oauth client events
+  Log.setLevel(Log.WARN);
+  Log.setLogger(console);
```
Comment (dinvlad) [inline src/index.jsx:138]: Please make sure we don't log the tokens accidentally, even if it's just via `console.log` :)

Comment (nmalfroy) [reply src/index.jsx:138]: haha oh totally.  This is still very much a draft

File: src/index.jsx [hunk 2]
```diff
@@ -103,27 +103,38 @@ function bootstrap() {
 function render(Component) {
   const root = document.getElementById('react');
   const { configuration } = store.getState();
-  const googleAuthority = 'https://accounts.google.com';
-  // TODO once we no longer query BQ directly, we'll no longer need the BQ scope
+  const isGoogleAuthority = configuration.configObject.authorityEndpoint.startsWith('https://accounts.google.com');
```
Comment (dinvlad) [inline src/index.jsx]: Which format does the new authority endpoint take that we can no longer use strict equality?

Comment (nmalfroy) [reply src/index.jsx]: Oh it could have a slash at the end

**Comment by nmalfroy** [comment]
That's

> Oh I see - the entire "user data" config (including refresh token) is removed upon logout right?

That's correct. When we do the 15 minute logout, it goes one step further and actually logs the user out of google

File: src/index.jsx (hunk 1 above)
Comment (snf2ye) [reply src/index.jsx:138]: Sanity check - Anything leftover from when this was a draft?

**Review by snf2ye — APPROVED** [review]

File: src/index.jsx (hunk 1 above)
Comment (samanehsan) [reply src/index.jsx:138]: ~Was this just for debugging?~ Oops I didn't see Shelby's comment until I reloaded the page!

**Comment by nmalfroy** [comment]
I left that there (it's super minimal logging) that way it's easier for future us to add more logging locally

**Review by samanehsan — APPROVED** [review]

**Pull request merged**